### ISSUE TITLE
Clean up Dumpglob conditionals from vernacentries

### ISSUE THREAD

**Issue by Alizter** [coq/coq#15538]
See discussion in https://github.com/coq/coq/pull/14740/files#r790329186

@herbelin can you edit this and add some more details?

**Comment by herbelin** [comment]
> @herbelin can you edit this and add some more details?

The point of my comment was that testing the activation of glob dumping at each call to Dumpglob was useless since the test was done again inside Dumpglob. Moreover, the situation is currently inconsistent with a couple of commands testing the activation and other not. This would be clearer with a uniform policy (i.e. leaving Dumpglob test itself). So, I suspect that we can remove all tests of Dumpglob activation in vernacentries.

I can do a quick PR if you want.

**Comment by ejgallego** [comment]
Note that the reason some conditionals may be in place is to avoid slowdowns as to compute the parameters of the `Dumpglob` call, thus a bit of care (or a thunk) may be needed.

**Comment by herbelin** [comment]
Then, it is also ok to leave things as they are (and at worst with a comment telling it is for efficiency).

**Comment by SkySkimmer** [comment]
>Note that the reason some conditionals may be in place is to avoid slowdowns as to compute the parameters of the Dumpglob call, thus a bit of care (or a thunk) may be needed. (And thunking itself requires care due to global state)

I don't think that makes sense, the parameters look trivial and anyway dumping is pretty much always on.

**Comment by ejgallego** [comment]
Indeed I didn't review if that's good or now, I'm just pointing out that it needs review.

In similar cases in past, I got bit by calls to the printer for example, which are super expensive.
